### PR TITLE
Remove TestPlanTarget migration

### PR DESCRIPTION
**Pull request by evmiguel**
This PR addresses: [ENV DATA: 3. Backend work to decouple Browser and AT from Test Plan](https://app.asana.com/0/1201709804960055/1202072717120766) and [ENV DATA: 4. Migrate existing data and mark the results with latest major version](https://app.asana.com/0/1201709804960055/1202072717120819)

The recommended sequence of folders to review are:

- migrations
- seeders
- models
- models/services
- resolvers
- tests
- scripts
- services

Running the changes:
```
yarn sequelize db:migrate;
yarn sequelize db:seed:all;
```

### CODE REVIEW

File: server/migrations/20220509140537-versionCollection.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+                "atId" = tpt."atId",
+                "browserId" = tpt."browserId"
+                FROM "TestPlanTarget" AS tpt 
+                WHERE tpr."testPlanTargetId" = tpt.id
+              `,
+                { transaction }
+            );
+
+            try {
+                await exec(
```
Comment (evmiguel) [inline server/migrations/20220509140537-versionCollection.js:55]: @alflennik @howard-e I included a pg_dump to save the data just in case something goes wrong. Do you think we need this? I'm stumped on the best way to handle rolling back the data for TestPlanTarget especially.

File: server/migrations/20220509140537-versionCollection.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            await queryInterface.removeColumn('TestPlanReport', 'atId', {
+                transaction
+            });
+            await queryInterface.removeColumn('TestPlanReport', 'browserId', {
+                transaction
+            });
+
+            try {
+                await exec(
+                    `psql ${process.env.PGDATABASE} < ${__dirname}/pg_dump_testplantarget_at_browser_version.sql`
```
Comment (evmiguel) [inline server/migrations/20220509140537-versionCollection.js:125]: Same thing here.

File: server/tests/integration/graphql.test.js [hunk 3]
```diff
@@ -298,6 +299,12 @@ describe('graphql', () => {
                                 test {
                                     id
                                 }
+                               # atVersion {
```
Comment (evmiguel) [inline server/tests/integration/graphql.test.js:302]: I'd like to put these changes in another PR.

File: server/tests/integration/graphql.test.js [hunk 4]
```diff
@@ -137,6 +141,8 @@ describe('graphql', () => {
         const excludedTypeNameAndField = [
             // Items formatted like this:
             // ['TestResult', 'startedAt'],
+            ['PopulatedData', 'atVersion'],
```
Comment (evmiguel) [inline server/tests/integration/graphql.test.js:144]: I need help with this / more time.

File: server/migrations/20220509140537-versionCollection.js (hunk 1 above)
Comment (howard-e) [reply server/migrations/20220509140537-versionCollection.js]: This seems like a well done way to me!

File: server/tests/integration/graphql.test.js (hunk 3 above)
Comment (howard-e) [reply server/tests/integration/graphql.test.js:302]: Perhaps we can remove the commented lines here for now then?

**Review by howard-e — APPROVED** [review]
I have been including changes here in #421 as we aim to keep the integration going as smoothly as we can. I can say that after testing these changes, everything is working as it should be.

@evmiguel thanks for this heavy lift!

File: server/models/services/AtService.js [hunk 5]
```diff
@@ -225,7 +225,7 @@ const getAtVersionByQuery = async (
 ) => {
     return ModelService.getByQuery(
         AtVersion,
-        { atId, name, releasedAt },
+        { atId, ...(name && { name }), ...(releasedAt && { releasedAt }) },
```
[the remark this answers is not in the document]
Comment (evmiguel) [reply server/models/services/AtService.js:228]: I was getting null responses from the query when at least one of those items were missing or when the three items were not in the query at the same time.

File: server/tests/integration/graphql.test.js (hunk 3 above)
Comment (evmiguel) [reply server/tests/integration/graphql.test.js:304]: I put a TODO there

**Review by alflennik — APPROVED** [review]
Looks great Erika, I'm really happy to see all the progress that's occurred when I was occupied in other workstreams. I'm happy to approve this! I would say don't merge it though because I want to build off of this when I add atVersionId and browserVersionId support, and feel like I might as well use this branch as a base.

File: server/models/services/AtService.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (howard-e) [reply server/models/services/AtService.js:228]: I see, thanks for clearing that up for me. Odd behavior indeed but this works!
